### PR TITLE
fix bug with json not updating yaml loading after leaving the page an…

### PR DESCRIPTION
**Pull request by jdsmithit**
fix bug with json not updating yaml loading after leaving the page and returning.

Also includes a refactor to IMO make the code a little more readable.

Fixes bug #273

### CODE REVIEW

File: src/app/component/circular-heatmap/circular-heatmap.component.ts [hunk 1]
```diff
@@ -58,124 +65,116 @@ export class CircularHeatmapComponent implements OnInit {
...
+
+                  if (localStorageData != null && localStorageData.length > 0) {
+                    this.YamlObject[allDimensionNames[d]][
+                      allSubDimensionInThisDimension[s]
+                    ][allActivityInThisSubDimension[a]]['teamsImplemented'] =
+                      this.getTeamImplementedFromJson(
+                        localStorageData,
+                        allActivityInThisSubDimension[a]
+                      );
+                  }
```
Comment (jdsmithit) [inline src/app/component/circular-heatmap/circular-heatmap.component.ts:139]: This is the main change everything else is refactoring, it keeps the Teams data in sync in the YAML Object

File: src/app/component/circular-heatmap/circular-heatmap.component.ts [hunk 2]
```diff
@@ -58,124 +65,116 @@ export class CircularHeatmapComponent implements OnInit {
...
-            var totalActivityTeams: number = 0;
-            tempData['Dimension'] = allDimensionNames[i];
-            tempData['SubDimension'] = allSubDimensionInThisDimension[j];
-            tempData['Level'] = 'Level ' + (l + 1);
-            for (var k = 0; k < allActivityInThisSubDimension.length; k++) {
+            var level = 'Level ' + (l + 1);
+            var activity: activitySchema[] = [];
+            var activityCompletionStatus: number = -1;
+
+            for (var a = 0; a < allActivityInThisSubDimension.length; a++) {
```
Comment (wurstbrot) [inline src/app/component/circular-heatmap/circular-heatmap.component.ts:102]: I personally would like to use a more speaking variable name like `activityIndex`. But I am ok with it as it is better than beforehand.

**Review by wurstbrot — APPROVED** [review]

**Comment by wurstbrot** [comment]
Great PR, thank you!

File: src/app/component/circular-heatmap/circular-heatmap.component.ts (hunk 2 above)
Comment (jdsmithit) [reply src/app/component/circular-heatmap/circular-heatmap.component.ts:102]: I agree but the linter pushed it to the next line and made it really hard to read ... so it was a little bit of a balancing act.

**Pull request merged**